test/source/tests/decrypt.ts

```diff
@@ -394,26 +428,73 @@ export const defineDecryptTests = (testVariant: TestVariant, testWithBrowser: Te
+        'president@forged.com');
+      const inboxPage = await browser.newPage(t, TestUrls.extension(`chrome/settings/inbox/inbox.htm?acctEmail=${acctEmail}&threadId=${threadId}`));
+      await inboxPage.waitAll('iframe', { timeout: 2 });
+      const urls = await inboxPage.getFramesUrls(['/chrome/elements/pgp_block.htm'], { sleep: 10, appearIn: 20 });
+      expect(urls.length).to.equal(1);
+      const url = urls[0].split('/chrome/elements/pgp_block.htm')[1];
+      await BrowserRecipe.pgpBlockVerifyDecryptedContent(t, browser, {
+        params: url,
+        content: ['How is my message signed?'],
+        signature: ['President@Forged.Com', 'matching signature']
+      });
+    }));
+
+    ava.default(`decrypt - missing pubkey in "incorrect message digest" scenario`, testWithBrowser('ci.tests.gmail', async (t, browser) => {
+      const msgId = '1766644f13510f58';
+      const acctEmail = 'ci.tests.gmail@flowcrypt.test';
+      const signerEmail = 'sender.for.refetch@domain.com';
+      const data = await GoogleData.withInitializedData(acctEmail);
+      const msg = data.getMessage(msgId)!;
+      const signature = Buf.fromBase64Str(msg!.raw!).toUtfStr()
+        .match(/\-\-\-\-\-BEGIN PGP SIGNATURE\-\-\-\-\-.*\-\-\-\-\-END PGP SIGNATURE\-\-\-\-\-/s)![0];
+      const params = `?frameId=none&account_email=${acctEmail}&senderEmail=${signerEmail}&msgId=${msgId}&message=Some%20corrupted%20message&signature=${encodeURIComponent(signature)}`;
+      // as the verification pubkey is not known, this scenario doesn't trigger message re-fetch
+      await BrowserRecipe.pgpBlockVerifyDecryptedContent(t, browser, { params, content: ['Some corrupted message'], signature: ["Missing pubkey 2864E326A5BE488A"] });
+    }));
+
+    ava.default('decrypt - re-fetch signed-only message from API on non-fatal verification error', testWithBrowser('compatibility', async (t, browser) => {
+      const msgId = '1754cfd1b2f1d6e5';
+      const acctEmail = 'flowcrypt.compatibility@gmail.com';
+      const signerEmail = 'dhartley@verdoncollege.school.nz';
+      const data = await GoogleData.withInitializedData(acctEmail);
+      const msg = data.getMessage(msgId)!;
+      const signature = Buf.fromBase64Str(msg!.raw!).toUtfStr()
+        .match(/\-\-\-\-\-BEGIN PGP SIGNATURE\-\-\-\-\-.*\-\-\-\-\-END PGP SIGNATURE\-\-\-\-\-/s)![0];
+      const params = `?frameId=none&account_email=${acctEmail}&senderEmail=${signerEmail}&msgId=${msgId}&message=Some%20corrupted%20message&signature=${encodeURIComponent(signature)}`;
+      // as the verification pubkey is retrieved from the attester, the incorrect message digest will trigger re-fetching from API
+      await BrowserRecipe.pgpBlockVerifyDecryptedContent(t, browser, {
+        params,
+        content: [], // todo: #4164 I would expect '1234' here
```

Comment:
Highlighting todo. We'll need a new message anyway

Reply:
I constructed a new message containing '1234'. Here my concern is whether the "slightly modified" originally loaded message should be updated (in the end we would like to show signature next to the really signed message, don't we?)

Reply:
Let's discuss this in main comments below